core/src/util/mod.rs

```diff
@@ -24,6 +31,68 @@ lazy_static! {
     static ref NON_ESCAPED_PERCENT: FancyRegex = FancyRegex::new(r"(?<!\\\\)%").unwrap();
 }
 
+#[repr(C)]
+pub struct BsonBuffer {
+    pub data: *const u8,
+    pub length: usize,
+    pub capacity: usize,
+}
+
+/// This function handles libmongosqltranslate runCommands. It takes in a `runCommand`,
+/// handles serializing it into a BSON byte vector, calls the libmongosqltranslate runCommand,
+/// deserializes the response, and returns either an error or a valid response
+/// for the given `runCommand`.
+pub(crate) fn libmongosqltranslate_run_command<T: CommandName + Serialize>(
+    command: Command<T>,
+) -> Result<CommandResponse> {
+    let library = get_mongosqltranslate_library().ok_or(Error::UnsupportedClusterConfiguration(
+        "Enterprise edition was detected, but libmongosqltranslate was not found.".to_string(),
+    ))?;
+
```

Comment:
If there is a generic From impl...

```suggestion
pub(crate) fn libmongosqltranslate_run_command<T: CommandName + Serialize>(
    command: impl<Into<Command<T>>,
) -> Result<CommandResponse> {
    let command = command.into();
    let library = get_mongosqltranslate_library().ok_or(Error::UnsupportedClusterConfiguration(
        "Enterprise edition was detected, but libmongosqltranslate was not found.".to_string(),
    ))?;

```